target/linux/realtek/files-5.10/drivers/i2c/busses/i2c-rtl9300.c

```diff
@@ -0,0 +1,484 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+#include <linux/module.h>
+#include <linux/of_platform.h>
+#include "i2c-rtl9300.h"
+
+#define REG(x)		(i2c->base + x + (i2c->scl_num ? 0x1c : 0))
+#define REG_MASK(clear, set, reg)	\
+			writel((readl(REG(reg)) & ~(clear)) | (set), REG(reg))
```

Comment:
I guess it lowers your line count, but I don't think it makes the code very easy to read, when you use macros like these.

Reply:
This is pretty standard fare in device drivers in the kernel. I have seen it a couple of times, e.g. with a quick grep:
pad_w32_mask() in  drivers/pinctrl/pinctrl-falcon.c
xway_stp_w32_mask in drivers/gpio/gpio-stp-xway.c 
snd_hdac_updatel in include/sound/hdaudio_ext.h
and I think I got it from John, who uses it extensively, e.g. here:
arch/mips/lantiq/falcon/sysctrl.c:#define sysctl_w32_mask(m, clear, set, reg)